autospec/buildreq.py

```diff
@@ -738,8 +783,23 @@ def scan_for_configure(dirn):
         if any(f.endswith(".go") for f in files):
             add_buildreq("buildreq-golang")
             buildpattern.set_build_pattern("golang", default_score)
-            if "go.mod" in files:
-                config.set_gopath = False
+
+        if "go.mod" in files:
```

Comment:
Goodness unfortunately this is pushing my memory but I do remember originally I had that condition set and I think I removed it due to this not being the case. I am not opposed to adding it back though as the sets of things supported by this change has decreased since then.